### ISSUE TITLE
Error: Connection Rejected due to Unacceptable BD_ADDR

### ISSUE THREAD

**Issue by simmeverschoren** [rnconrad/WiimoteEmulator#12]
Thank you for your response on my previous issue and for pushing a new version! I have some trouble making it work with my Raspberry Pi 3B+ (Raspberry Pi OS) and Wii.

To resolve errors regarding bluez 5.50 and SDP, I have to run the bluetooth daemon in compatibility mode following [this answer on StackExchange](https://raspberrypi.stackexchange.com/a/42262). The errors were:
```
failed to get sdp server session
failed to add Wiimote SDP record
```

While running the emulator, the device settings appear to be properly set up: [wmemulator_settings.txt](https://github.com/rnconrad/WiimoteEmulator/files/5906968/wmemulator_settings.txt). I should mention that my bluetooth adapter requires to be unplugged and replugged for the new bluetooth address to be written. Doing this while running the emulator obviously doesn't work so I unplug the adapter in between emulator sessions.

Upon trying to connect the Wii, hcidump reports:
```
$ hcidump
HCI sniffer - Bluetooth packet analyzer ver 5.40
device: hci0 snap_len: 1500 filter: 0xffffffff
> HCI Event: Connect Request (0x04) plen 10
    bdaddr 00:22:D7:61:EF:AD class 0x000448 type ACL
< HCI Command: Reject Connection Request (0x01|0x000a) plen 7
    bdaddr 00:22:D7:61:EF:AD reason 0x0f
    Reason: Connection Rejected due to Unacceptable BD_ADDR
> HCI Event: Command Status (0x0f) plen 4
    Reject Connection Request (0x01|0x000a) status 0x00 ncmd 1
> HCI Event: Connect Complete (0x03) plen 11
    status 0x0f handle 43 bdaddr 00:22:D7:61:EF:AD type ACL encrypt 0x00
    Error: Connection Rejected due to Unacceptable BD_ADDR
```
- Logs from hcidump: [hcidump_logs.txt](https://github.com/rnconrad/WiimoteEmulator/files/5907372/hcidump_logs.txt).
- Logs from bluetoothd: [bluetoothd_logs.txt](https://github.com/rnconrad/WiimoteEmulator/files/5907371/bluetoothd_logs.txt).

Similar results pop up on Ubuntu 20.04 using bluez 5.53.

Trying with bluez 4.101 results in an authentication failure because of the PIN code.
- Logs from hcidump using bluez 4.101: [hcidump_4.101_logs.txt](https://github.com/rnconrad/WiimoteEmulator/files/5907373/hcidump_4.101_logs.txt).
- Logs from bluetoothd using bluez 4.101: [bluetoothd_4.101_logs.txt](https://github.com/rnconrad/WiimoteEmulator/files/5907370/bluetoothd_4.101_logs.txt).

Any help would be appreciated.

**Comment by rnconrad** [comment]
It seems there are still some unresolved issues with bluez 5.5x after all, sorry for the confusion there.

For now I'd suggest using the [custom build script](https://github.com/rnconrad/WiimoteEmulator/blob/master/CustomBuild.md), which builds a local, patched version of bluez 4.101 that you can use for the emulator. I just tested this on a Raspberry Pi 3B+ with the latest RPi OS and it works for me.

Also, the emulator is currently unable to automatically set the device address for the adapter on the 3B+ (and some others), so you'll have to manually set it to any Nintendo address (as you mentioned). The emulator no longer quits when it fails to do this, it just emits a warning so the address can be set separately.

Edit: note that I've pushed changes to the build script recently, be sure to use those or else it probably won't work.

**Comment by simmeverschoren** [comment]
Works great! Thank you very much!
